src/main/java/org/cryptomator/ui/keyloading/hub/RegisterDeviceController.java

```diff
@@ -49,18 +59,34 @@ public class RegisterDeviceController implements FxController {
 	public TextField deviceNameField;
 
 	@Inject
-	public RegisterDeviceController(@KeyLoading Stage window, ExecutorService executor, HubConfig hubConfig, @Named("deviceId") String deviceId, DeviceKey deviceKey, CompletableFuture<JWEObject> result, @Named("bearerToken") AtomicReference<String> bearerToken) {
+	public RegisterDeviceController(@KeyLoading Stage window, ExecutorService executor, HubConfig hubConfig, @Named("deviceId") String deviceId, DeviceKey deviceKey, CompletableFuture<JWEObject> result, @Named("bearerToken") AtomicReference<String> bearerToken, @FxmlScene(FxmlFile.HUB_REGISTER_SUCCESS) Lazy<Scene> registerSuccessScene, @FxmlScene(FxmlFile.HUB_REGISTER_FAILED) Lazy<Scene> registerFailedScene) {
 		this.window = window;
 		this.hubConfig = hubConfig;
 		this.deviceId = deviceId;
 		this.keyPair = Objects.requireNonNull(deviceKey.get());
 		this.result = result;
 		this.bearerToken = Objects.requireNonNull(bearerToken.get());
+		this.registerSuccessScene = registerSuccessScene;
+		this.registerFailedScene = registerFailedScene;
 		this.jwt = JWT.decode(this.bearerToken);
 		this.window.addEventHandler(WindowEvent.WINDOW_HIDING, this::windowClosed);
 		this.httpClient = HttpClient.newBuilder().executor(executor).build();
 	}
 
+	public void initialize() {
+		deviceNameField.setText(System.getProperty("user.name") + "-" + determineHostname());
+	}
+
+	private String determineHostname() {
+		try (var inputStream = Runtime.getRuntime().exec("hostname").getInputStream(); //
```

Comment:
spawning a new process ist costly and error-prone - why not use `InetAddress.getLocalHost().getHostName()`?

Reply:
Went into this rabbid hole: https://stackoverflow.com/questions/7348711/recommended-way-to-get-hostname-in-java/40702767#40702767

It is also linked in the other SO Thread.

Reply:
I see. I would prefer a `hostname -s` though, no need to include the fqdn.

This does block the UI thread, doesn't it? Does the process reliably end within a short amount of time?

Reply:
After a short discussion: Spawning the process is too much hassle, since its behaviour over different systems is not clear. The InetAddress approach is used and if this fails, we just an empty string in the text box.

Changed inhttps://github.com/cryptomator/cryptomator/pull/2319/commits/0e853b25f7a8927c5a0e17ecd299422e395d64d5